lib/deePool.src.js

```diff
@@ -29,25 +29,25 @@
 		// ******************************
 
 		function use() {
-      if (freePoolEnd == -1) {
-        if (freePool.length == 0) {
-          grow(5);
-        } else {
-          grow();
-        }
-      }
-      return freePool[freePool.length - (freePoolEnd--)];
+			if (freePoolEnd == -1) {
+				if (freePool.length == 0) {
+					grow(5);
+				} else {
+					grow();
+				}
+			}
+			return freePool[freePool.length - 1 - (freePoolEnd--)];
```

Comment:
Couldn't freePoolEnd be renamed freePoolIndex, start out at 0, increase on use(), decrease on recycle(), and just be checked against freePool.length instead of -1 to know when we need to grow?